subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -347,40 +351,19 @@ impl Session<Error> for ElectrumSession {
         )
         .ok_or(Error::InvalidMnemonic)?;
         let secp = Secp256k1::new();
-        let xprv =
-            ExtendedPrivKey::new_master(bitcoin::network::constants::Network::Testnet, &seed)?;
 
-        // BIP44: m / purpose' / coin_type' / account' / change / address_index
-        // coin_type = 0 bitcoin, 1 testnet, 1776 liquid bitcoin as defined in https://github.com/satoshilabs/slips/blob/master/slip-0044.md
-        // slip44 suggest 1 for every testnet, so we are using it also for regtest
-        let coin_type: u32 = match self.network.id() {
-            NetworkId::Bitcoin(bitcoin_network) => match bitcoin_network {
-                bitcoin::Network::Bitcoin => 0,
-                bitcoin::Network::Testnet
-                | bitcoin::Network::Regtest
-                | bitcoin::Network::Signet => 1,
-            },
-            NetworkId::Elements(elements_network) => match elements_network {
-                ElementsNetwork::Liquid => 1776,
-                ElementsNetwork::ElementsRegtest => 1,
-            },
-        };
-        // since we use P2WPKH-nested-in-P2SH it is 49 https://github.com/bitcoin/bips/blob/master/bip-0049.mediawiki
-        let path_string = format!("m/49'/{}'/0'", coin_type);
-        info!("Using derivation path {}/0|1/*", path_string);
-        let path = DerivationPath::from_str(&path_string)?;
-        let xprv = xprv.derive_priv(&secp, &path)?;
-        let xpub = ExtendedPubKey::from_private(&secp, &xprv);
-
-        let wallet_id = self.network.unique_id(&xpub);
-        let sync_interval = self.network.sync_interval.unwrap_or(7);
+        let master_xprv =
+            ExtendedPrivKey::new_master(bitcoin::network::constants::Network::Testnet, &seed)?;
+        let master_xpub = ExtendedPubKey::from_private(&secp, &master_xprv);
 
         let master_blinding = if self.network.liquid {
             Some(asset_blinding_key_from_seed(&seed))
         } else {
             None
         };
 
+        let wallet_id = self.network.unique_id(&master_xpub);
```

Comment:
This is a breaking change.
Before wallet id was derived from `M/49'/cointype'/0'`, now from `M`.
This for instance would loose users memos.
I think that we ultimately want to derive the wallet id from from the master xpub, but IMHO such a (breaking) change should belong to its own commit.
An approach that we can take is implemented in https://github.com/Blockstream/gdk/pull/98, where we do a breaking change for all wallets, but we attempt to recover data store in the dir the Aqua used.

Reply:
This and the other breaking change:

```
-            Err(_) => Arc::new(RwLock::new(StoreMeta::new(
-                &path,
-                xpub,
-                master_blinding.clone(),
-                self.network.id(),
-            )?)),
+            Err(_) => Arc::new(RwLock::new(StoreMeta::new(&path, master_xpub, self.network.id())?)),
```
Should be fixed in #98 which is based on this